middleware/file/source/manager/admin/cli.cpp

```diff
@@ -0,0 +1,174 @@
+               {
+                  auto requests( const manager::admin::model::State& state)
+                  {
+                     const auto color = common::terminal::output::directive().porcelain() ? 
+                        common::terminal::color::yellow : 
+                        common::terminal::color::no_color;
+
+                     return common::terminal::format::formatter< manager::admin::model::Request>::construct(
+                        common::terminal::format::column( "process", []( const auto& r){ return r.process;}, color),
+                        common::terminal::format::column( "gtrid", []( const auto& r){ return r.gtrid;}, color),
+                        common::terminal::format::column( "stage", []( const auto& r){ return description( r.stage);}, color),
+                        common::terminal::format::column( "path", []( const auto& r){ return r.path;}, color));
+                  }
+               } // format
+
+            } // detail
+
+            namespace list
+            {
+               namespace reservations
+               {
+                  auto option()
+                  {
+                     auto invoke = []()
+                     {
+                        auto state = detail::call::state();
+
+                        auto formatter = detail::format::requests( state);
+
+                        formatter.print( std::cout, state.requests);
+                     };
+
+                     return argument::Option{
+                        std::move( invoke),
+                        { "-r", "--list-reservations"},
+                        "list information of files currently reserved"
+                     };
+                  }
+               } // files
+
+            } // list
+
+            namespace assets
+            {
+               
+               namespace recovery
+               {
+                  namespace detail
+                  {
+                     auto invoke( const manager::admin::model::recovery::Directive directive)
+                     {
+                        return [directive]( common::transaction::global::ID gtrid, std::vector< common::transaction::global::ID> gtrids)
+                        {
+                           gtrids.insert( std::begin( gtrids), std::move( gtrid));
+                           const auto recovered = local::detail::call::recover( std::move( gtrids), directive); 
+
+                           for( const auto& gtrid : recovered)
+                           {
+                              common::log::line( std::cout, gtrid);
+                           }
+                        };
+                     }
+
+                     auto complete()
+                     {
+                        return []( bool help, auto values) -> std::vector< std::string>
+                        {
+                           if( help)
+                              return { "<gtrid>"};
+
+                           return { "<value>"};
```

Comment:
You could call state, and return all gtrid that the manager knows about. user can then tap-complete gtrids